pysb/importers/sbml.py

```
@@ -0,0 +1,129 @@
+
+    p = subprocess.Popen(sbmltrans_args,
+                         cwd=os.getcwd(),
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.PIPE)
+
+    if verbose:
+        for line in iter(p.stdout.readline, b''):
+            print(line, end="")
+    (p_out, p_err) = p.communicate()
+    if p.returncode:
+        raise SbmlTranslationError(p_out.decode('utf-8') + "\n" +
+                                   p_err.decode('utf-8'))
+
+    return output_file
+
+
+def model_from_sbml(filename, cleanup=True, **kwargs):
+    """
+    Create a PySB Model object from an Systems Biology Markup Language (SBML)
+    file, using BioNetGen's
+    `sbmlTranslator <http://bionetgen.org/index.php/SBML2BNGL>`_,
+    which can attempt to extrapolate higher-level (rule-based) structure
+    from an SBML source file (argument atomize=True). The model is first
+    converted into BioNetGen language by sbmlTranslator, then PySB's
+    :class:`BnglBuilder` class converts the BioNetGen language model into a
+    PySB Model.
+
+    Limitations
+    -----------
+
+    Read the `sbmlTranslator documentation
+    <http://bionetgen.org/index.php/SBML2BNGL>`_ for further information on
+    sbmlTranslator's limitations.
+
+    Parameters
+    ----------
+    filename :
+        A Systems Biology Markup Language .sbml file
+    cleanup : bool
+        Delete temporary directory on completion if True. Set to False for
+        debugging purposes.
+    **kwargs: kwargs
+        Keyword arguments to pass on to :func:`SbmlBuilder.sbml_translator`
+    """
+    tmpdir = tempfile.mkdtemp()
+    verbose = kwargs.get('verbose', False)
+    if verbose:
+        print("Performing SBML to BNGL translation in temporary "
+              "directory %s" % tmpdir)
+    try:
+        bngl_file = os.path.join(tmpdir, 'model.bngl')
+        sbml_translator(filename, bngl_file, **kwargs)
+        return model_from_bngl(bngl_file)
+    finally:
```

This silently swallows all exceptions... is that what was intended?